boards/st/stm32h7s78_dk/stm32h7s78_dk.dts

```diff
@@ -351,3 +363,15 @@ zephyr_udc0: &usb2 {};
 	def-back-color-green = <0xFF>;
 	def-back-color-blue = <0xFF>;
 };
+
+&sdmmc1 {
+	pinctrl-0 = <&sdmmc1_d0_pc8 &sdmmc1_d1_pc9
+		     &sdmmc1_d2_pc10 &sdmmc1_d3_pc11
+		     &sdmmc1_ck_pc12 &sdmmc1_cmd_pd2>;
+	pinctrl-names = "default";
+	clk-div = <13>;
+	bus-width = <4>;
+	cd-gpios = <&gpiom 14 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
```

Comment:
It should be `GPIO_ACTIVE_LOW`.
I have the stm32h7s78_dk and I tested your branch. It works with `GPIO_ACTIVE_LOW`.